packages/gatsby/src/utils/babel-loader-helpers.ts

```diff
@@ -70,17 +94,17 @@ const prepareOptions = (babel, options = {}, resolve = require.resolve) => {
   )
 
   // Go through babel state and create config items for presets/plugins from.
-  const reduxPlugins = []
-  const reduxPresets = []
-  pluginBabelConfig.stages[stage].plugins.forEach(plugin => {
+  const reduxPlugins: PluginItem[] = []
+  const reduxPresets: PluginItem[] = []
+  pluginBabelConfig.stages[stage!].plugins.forEach(plugin => {
```

Comment:
If you've typed the options, and the stage key as optional you'll need a runtime null check for `stage` rather than this assertion.